core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -200,17 +176,50 @@ class CassandraJournal(cfg: Config) extends AsyncWriteJournal with CassandraReco
               case write :: remainder => writeMessages(write).flatMap(result => rec(remainder, result :: acc))
               case Nil                => Future.successful(acc.reverse)
             }
+
           rec(groups, Nil)
         }
 
-      result.onComplete { _ =>
+      val resultWithNotification = result.flatMap({ _ =>
+        val messagesByTag: Map[String, Seq[Serialized]] = serialized
+          .flatMap(_.payload)
+          .flatMap(s => s.tags.map((_, s)))
+          .groupBy(_._1).mapValues(_.map(_._2))
+
+        // todo make configurable and/or cirvuit break tag writes
+        implicit val timeout = Timeout(1.second)
+        val acks = messagesByTag.map {
+          case (tag, writes) =>
+            val tagWriter: ActorRef = tagActors.computeIfAbsent(tag, new JFunction[String, ActorRef] {
+              def apply(t: String): ActorRef = {
+                val tagSession = new TagWriterSession(
+                  tag,
+                  preparedWriteToTagView,
+                  session.executeWrite,
+                  session.selectResultSet,
+                  preparedWriteToTagProgress,
+                  preparedWriteToTagFirstBucket,
+                  preparedSelectTagProgress
+                )
+                context.actorOf(
```

Comment:
This is executing in a future callback so this is not allowed. I think it would be easiest to just send the `TagWrites` message to `self`, which will create-and-delegate to the tagWriter actor

Reply:
or have an addition parent actor for all tagWriter actor and just delegate to that one from here

Reply:
+1

Reply:
I've sent it to self for now